pkg/controller/installation/products/monitoring/reconciler.go

```diff
@@ -218,3 +283,62 @@ func (r *Reconciler) createResource(inst *v1alpha1.Installation, resourceName st
 
 	return resource, nil
 }
+
+// Read the credentials of the Prometheus instance in the openshift-monitoring
+// namespace from the grafana datasource secret
+func (r *Reconciler) readFederatedPrometheusCredentials(ctx context.Context, serverClient pkgclient.Client) (*monitoring_v1alpha1.GrafanaDataSourceSecret, error) {
+	secret := &v12.Secret{}
+
+	selector := client.ObjectKey{
+		Namespace: openshiftMonitoringNamespace,
+		Name:      grafanaDataSourceSecretName,
+	}
+
+	err := serverClient.Get(ctx, selector, secret)
+	if err != nil {
+		return nil, err
+	}
+
+	prometheusConfig := secret.Data[grafanaDataSourceSecretKey]
+	datasources := monitoring_v1alpha1.GrafanaDataSourceSecret{}
+
+	err = json.Unmarshal(prometheusConfig, &datasources)
+	if err != nil {
+		return nil, err
+	}
+
+	return &datasources, err
+}
+
+// Populate the extra params for templating
+func (r *Reconciler) populateParams(ctx context.Context, inst *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	// Obtain the prometheus credentials from openshift-monitoring
+	datasources, err := r.readFederatedPrometheusCredentials(ctx, serverClient)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	if len(datasources.DataSources) < 1 {
+		return v1alpha1.PhaseFailed, errors.New("cannot obtain prometheus credentials")
+	}
+
+	// Obtain the 3scale config and namespace
+	threeScaleConfig, err := r.ConfigManager.ReadThreeScale()
```

Comment:
Following on from the discussion with regard to monitoring resources and where and when they get installed, does this need now to be handled in the threescale reconciler instead?

Reply:
yes, I was just thinking about that.
Let's discuss it with @obrienrobert tomorrow?

Reply:
@davidkirwan @matskiv @obrienrobert The monitoring operator does only allow for one additional scrape config secret (see [here](https://github.com/integr8ly/application-monitoring-operator/blob/46c0b4c13fe4d216ae7475f5e8462e647889dd4d/pkg/controller/applicationmonitoring/applicationmonitoring_controller.go#L554)) and we concatenate all the jobs into this one secret. So in my opinion it makes sense to generate it here.